shm/inc/Game.hpp

```diff
@@ -24,7 +30,10 @@ class Game {
     std::unique_ptr<Player> player_;
     std::unique_ptr<Time> time_;
     std::unique_ptr<Ship> ship_;
+    std::unique_ptr<Map> map_;
+    std::map<size_t, std::pair<size_t, size_t>> island;
```

Comment:
no need for additional memory usage - container of islands already exists in Map class